### PR TITLE
Fix refract mson

### PR DESCRIPTION
**Pull request by klokane**
- fix support for samples and defaults on Named Types && Array
 - refactoring Mson(Property|Value)ToRefract

### CODE REVIEW

File: src/RefractAST.cc [hunk 1]
```diff
@@ -194,47 +194,75 @@ namespace drafter
...
     struct ExtractTypeSection
     { // This will handle primitive elements
-        const mson::TypeSection& ts;
+        std::vector<typename T::ValueType> values;
+        TypeSectionData& data;
 
-        ExtractTypeSection(const mson::TypeSection& v) : ts(v)
-        {
-        }
+        template <typename U, bool dummy = true>
```
Comment (klokane) [inline src/RefractAST.cc]: This `dummy` param is required as workaround for inner template specialization
will add explaining comment

File: src/RefractAST.cc [hunk 2]
```diff
@@ -194,47 +194,75 @@ namespace drafter
...
+
+    struct TypeSectionData {
+        RefractElements defaults;
+        RefractElements samples;
+        std::vector<std::string> descriptions;
+    };
+
+    template <typename T>
     struct ExtractTypeSection
     { // This will handle primitive elements
```
Comment (pksunkara) [inline src/RefractAST.cc]: The previous PR had a comment on this line saying that this comment needs to be removed. I haven't looked into it yet but wanted to mention it.

Comment (klokane) [inline src/RefractAST.cc]: thx for remind

**Comment by klokane** [comment]
Fixed

**Comment by klokane** [comment]
@pksunkara what is status of review?

**Comment by pksunkara** [comment]
Reviewed and lgtm. I just have one concern which we need to write test cases for after these PRs are merged.

**Pull request merged**